src/manage_sql.c

```diff
@@ -16695,19 +16725,19 @@ check_db (int check_encryption_key)
   create_tables ();
   check_db_sequences ();
   set_db_version (GVMD_DATABASE_VERSION);
-  check_db_roles ();
-  check_db_nvt_selectors ();
+  check_db_roles (avoid_db_check_inserts);
+  check_db_nvt_selectors (avoid_db_check_inserts);
   check_db_nvts ();
-  check_db_port_lists ();
+  check_db_port_lists (avoid_db_check_inserts);
   clean_auth_cache ();
-  if (check_db_scanners ())
+  if (check_db_scanners (avoid_db_check_inserts))
     goto fail;
-  if (check_db_report_formats ())
+  if (check_db_report_formats (avoid_db_check_inserts))
     goto fail;
   if (check_db_report_formats_trash ())
     goto fail;
-  check_db_permissions ();
-  check_db_settings ();
+  check_db_permissions (avoid_db_check_inserts);
+  check_db_settings (avoid_db_check_inserts);
```

Comment:
Some of the above functions (`check_db_roles`, `check_db_nvt_selectors`, `check_db_scanners`, `check_db_permissions`, `check_db_settings`) are not called elsewhere and they only return when `avoid_db_check_inserts = 1`. Isn't it sufficient to wrap an if around them ?